app/controllers/api/v1/reference/suppliers_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Reference
+      class SuppliersController < ApiController
+        def index
+          suppliers = Supplier.all
+          render json: suppliers
+        end
+
+        def show
+          supplier = find_supplier
+          raise ::ActiveRecord::RecordNotFound, "Couldn't find Supplier with #{params[:id]}" if supplier.nil?
+
+          render json: supplier, status: 200
+        end
+
+        private
+
+        def find_supplier
+          if ::UUID.validate(params[:id])
+            Supplier.find(params[:id])
+          else
+            Supplier.find_by(key: params[:id])
+          end
+        end
```

Comment:
Hi @ngw, are we moving away from the service objects approach (e.g. https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/services/locations/finder.rb)?

Reply:
Also, are we sure we need to look this up abiguously? I've asked on https://dsdmoj.atlassian.net/browse/P4-730.

Reply:
About service objects no, we're not moving away, I just didn't think about it at that moment because we mostly don't use finders and query directly the model (see for example `find_location`). I guess I'll wait to see what happens with the find by key or id which is actually the only logic that could go in a service object and eventually refactor. Good enough?

Reply:
Sounds good to me :+1: